agent/xds/testdata/listeners/mesh-gateway-custom-addresses.golden

```diff
@@ -168,6 +216,22 @@
             }
           ]
         },
+        {
+          "filterChainMatch": {
+            "serverNames": [
+              "*.dc4.internal.11111111-2222-3333-4444-555555555555.consul"
+            ]
+          },
+          "filters": [
+            {
+              "name": "envoy.tcp_proxy",
+              "config": {
+                  "cluster": "dc4.internal.11111111-2222-3333-4444-555555555555.consul",
+                  "stat_prefix": "mesh_gateway_remote_foo_dc4_tcp"
+                }
+            }
+          ]
+        },
```

Comment:
Why would these change for mesh gateways? This doesn't seem related to the issue being fixed here. Is this an unintended side effect of the logic change?

Reply:
I added dc4 to the default test snapshot for mesh gateways here:
https://github.com/hashicorp/consul/pull/7999/files#diff-f8966ffd7dc2c27418ed8181bcedeecdR1390

Now in all mesh gateway tests we create an additional `filterChainMatch` for traffic to dc4.